src/components/OptionsModal/GeneralOptionsPane.js

```diff
@@ -79,9 +79,17 @@ function GeneralOptionsPane({ options, updateOption }) {
                   </ControlLabel>
                   <FormControl
                     type="number"
+                    min="0"
                     value={options.get('historySize', DEFAULT_HISTORY_SIZE)}
                     onChange={e => {
-                      updateOption('historySize', e.target.value);
+                      try {
+                        let value = parseInt(e.target.value, 10);
+                        if (value < 0) value = 0; // prevents manually typing negative value
+                        updateOption('historySize', value);
+                      } catch (parseIntException) {
+                        // Ignore any attempts to type an invalid value,
+                        // like letters, or a negative sign
+                      }
```

Comment:
This is great! Let me suggest simplifying it slightly though. We can use `e.target.valueAsNumber` as `FormControl` is a number input (`type=number`) - so it exposes that special property which returns a number directly.

In theory a number input should not dispatch any update events unless the input is a valid number, but in practice you may get `NaN` back sometimes like when you try to clear the input so it never hurts to be safe. `isNaN` should guard against anything that `valueAsNumber` can return that we don't want.

```suggestion
                      const value = e.target.valueAsNumber;
                      // Prevent invalid input
                      if (isNaN(value)) return;
                      updateOption('historySize', value >= 0 ? value : 0);
```

Reply:
Cool, I didn't know `e.target.valueAsNumber` existed. Is that native JS, or is React adding it as a handy feature? Makes this code much simpler.